service/storage_proxy.cc

```diff
@@ -683,23 +677,7 @@ class storage_proxy::remote {
         }
         future<rpc::tuple<query::result_digest, api::timestamp_type, cache_temperature, std::optional<full_position>>> f = co_await coroutine::as_future(p->query_result_local_digest(std::move(s), cmd, std::move(pr2.first), trace_state_ptr, timeout, da, rate_limit_info));
         tracing::trace(trace_state_ptr, "read_digest handling is done, sending a response to /{}", src_ip);
-
-        using final_tuple_type = rpc::tuple<query::result_digest, long, cache_temperature, replica::exception_variant, std::optional<full_position>>;
-
-        if (!f.failed()) {
-            auto&& [d, t, c, p] = f.get();
-            co_return final_tuple_type(d, t, std::move(c), replica::exception_variant(), std::move(p));
-        }
-
-        std::exception_ptr eptr = f.get_exception();
-        if (p->features().typed_errors_in_read_rpc) {
-            replica::exception_variant ex = replica::try_encode_replica_exception(eptr);
-            if (ex) {
-                co_return final_tuple_type(std::tuple(query::result_digest(), api::missing_timestamp, cache_temperature::invalid(), replica::exception_variant(std::move(ex)), std::nullopt));
```

Comment:
The new code will return `final_tuple_type{}`, I see that the `long` will be different (it won't be `api::missing_timestamp` but `0`) - won't that be a problem?

Reply:
> won't that be a problem?

no, these values will be just dropped on the call site in case of exception, e.g.:
```
auto&& [result, hit_rate, opt_exception] = co_await ser::storage_proxy_rpc_verbs::send_read_data(&_ms, addr, timeout, cmd, pr, digest_algo, rate_limit_info, fence);
if (opt_exception.has_value() && *opt_exception) {
    co_await coroutine::return_exception_ptr((*opt_exception).into_exception_ptr());
}
```